playwright/pages/customersPage.ts

```diff
@@ -1,12 +1,87 @@
-import type { Page } from "@playwright/test";
+import { URL_LIST } from "@data/url";
+import { Page, expect } from "@playwright/test";
+import { AddressForm, customerAddress } from "@forms/addressForm";
+import { BasePage } from "@pages/basePage";
+import { DeleteDialog } from "@dialogs/deleteDialog";
+import { IssueGiftCardDialog } from "@dialogs/issueGiftCardDialog";
+import { AddressDialog } from "@dialogs/addressDialog";
 
-export class CustomersPage {
-  readonly page: Page;
+export class CustomersPage extends BasePage {
+  readonly addressForm: AddressForm;
+  readonly deleteDialog: DeleteDialog;
+  readonly issueGiftCardDialog: IssueGiftCardDialog;
+  readonly addressDialog: AddressDialog;
 
   constructor(
     page: Page,
     readonly createCustomerButton = page.getByTestId("create-customer"),
+    readonly customerFirstNameInput = page
+      .getByTestId("customer-first-name")
+      .locator("input"),
+    readonly customerLastNameInput = page
+      .getByTestId("customer-last-name")
+      .locator("input"),
+    readonly customerEmailInput = page
+      .getByTestId("customer-email")
+      .locator("input"),
+    readonly customerNoteInput = page
+      .getByTestId("customer-note")
+      .locator("textarea[name='note']"),
+    readonly saveButton = page.getByTestId("button-bar-confirm"),
+    readonly deleteButton = page.getByTestId("button-bar-delete"),
+    readonly issueNewGiftCardButton = page.getByTestId("issue-new-gift-card"),
+    readonly menageAddressButton = page.getByTestId("manage-addresses"),
+    readonly showMoreMenuButton = page.getByTestId("show-more-button"),
+    readonly editAddressButton = page.getByTestId("edit-address"),
   ) {
-    this.page = page;
+    super(page);
+    this.addressForm = new AddressForm(page);
+    this.deleteDialog = new DeleteDialog(page);
+    this.issueGiftCardDialog = new IssueGiftCardDialog(page);
+    this.addressDialog = new AddressDialog(page);
+  }
+
+  async goToCustomersListView() {
+    await this.page.goto(URL_LIST.customers);
+  }
+  async gotoCustomerDetailsPage(customerId: string) {
+    const existingCustomerUrl = `${URL_LIST.customers}${customerId}`;
+    console.info(`Navigating to existing customer: ${existingCustomerUrl}`);
+    await this.page.goto(existingCustomerUrl);
+    await expect(this.pageHeader).toBeVisible({ timeout: 10000 });
```

Comment:
question: are you show that when the page header appears all data will be ready? IMO we should wait for something that give us information that data from API has been loaded, the page header appears before we get data from API.